core/node/state_keeper/src/updates/mod.rs

```diff
@@ -167,19 +176,23 @@ impl UpdatesManager {
 
     /// Pushes a new L2 block with the specified timestamp into this manager. The previously
     /// held L2 block is considered sealed and is used to extend the L1 batch data.
-    pub fn push_l2_block(&mut self, l2_block_params: L2BlockParams) {
+    pub fn push_l2_block(&mut self) {
         let new_l2_block_updates = L2BlockUpdates::new(
-            l2_block_params.timestamp,
+            self.next_l2_block_param.timestamp,
             self.l2_block.number + 1,
             self.l2_block.get_l2_block_hash(),
-            l2_block_params.virtual_blocks,
+            self.next_l2_block_param.virtual_blocks,
             self.protocol_version,
         );
         let old_l2_block_updates = std::mem::replace(&mut self.l2_block, new_l2_block_updates);
         self.l1_batch
             .extend_from_sealed_l2_block(old_l2_block_updates);
     }
 
+    pub fn set_next_l2_block_parameters(&mut self, l2_block_param: L2BlockParams) {
```

Comment:
Bikeshedding: I'd rename this method to `set_next_l2_block_params()` to align with the field / type naming.